Assets/MixedRealityToolkit/Interfaces/SceneSystem/IMixedRealitySceneSystem.cs

```diff
@@ -0,0 +1,245 @@
+        /// </summary>
+        Action<string> OnWillLoadScene { get; set; }
+
+        /// <summary>
+        /// Called when scene has been loaded, activated and is valid.
+        /// Called for all scene types (content, lighting & manager)
+        /// Includes scene name
+        /// </summary>
+        Action<string> OnSceneLoaded { get; set; }
+
+        /// <summary>
+        /// Called just before a scene will be unloaded
+        /// Called for all scene types (content, lighting & manager)
+        /// Includes scene name
+        /// </summary>
+        Action<string> OnWillUnloadScene { get; set; }
+
+        /// <summary>
+        /// Called when scene has been unloaded
+        /// Called for all scene types (content, lighting & manager)
+        /// Includes scene name
+        /// </summary>
+        Action<string> OnSceneUnloaded { get; set; }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// True if the scene system is loading or unloading content scenes.
+        /// Manager and lighting scenes are ignored.
+        /// </summary>
+        bool SceneOperationInProgress { get; }
+
+        /// <summary>
+        /// Progress of the current scene operation, from 0-1.
+        /// A scene operation may include multiple concurrently loaded scenes.
+        /// </summary>
+        float SceneOperationProgress { get; }
+
+        /// <summary>
+        /// True when content has been loaded with an activation token and AllowSceneActivation has not been set to true.
+        /// Useful for existing entities that shouldn't act until a newly loaded scene is actually activated.
+        /// </summary>
+        bool WaitingToProceed { get; }
+
+        /// <summary>
+        /// Name of the currently loaded lighting scene.
+        /// If a transition is in progress, this reports the target lighting scene we're transitioning to.
+        /// </summary>
+        string ActiveLightingScene { get; }
+
+        #endregion
+
+        #region Scene Operations
+
+        /// <summary>
+        /// Async method to load the scenes by name.
+        /// If a scene operation is in progress, no action will be taken.
+        /// </summary>
+        /// <param name="scenesToLoad">Names of content scenes to load. Invalid scenes will be ignored.</param>
+        /// <param name="mode">Additive mode will load the content additively. Single mode will first unload all loaded content scenes first.</param>
+        /// <param name="activationToken">
+        /// Optional token for manual scene activation. Useful for loading screens and multiplayer.
+        /// If not null, operation will wait until activationToken's AllowSceneActivation value is true before activating scene objects.
+        /// </param>
+        /// <returns>Task</returns>
+        Task LoadContent(IEnumerable<string> scenesToLoad, LoadSceneMode mode = LoadSceneMode.Additive, SceneActivationToken activationToken = null);
```

Comment:
Can you give an explanation of why I would want to use MRTK scene system instead of Unity APIs to load and unload scenes?

Reply:
Sure. First, unity's basic scene manager is full of [little gotchas](https://issuetracker.unity3d.com/issues/scenemanager-dot-getscenebybuildindex-dot-name-returns-an-empty-string-if-scene-is-not-loaded) that nibble away at productivity and cause hard-to-diagnose bugs. This service protects you from those.

Second, keeping a manager scene loaded while you additively load / unload other scenes (which is what both we and Unity recommend you do) can't be done without some sort of scene management system written on top of Unity's basic system. Using our service means you won't have to reinvent that wheel for every project.